### ISSUE TITLE
Additional cmd.exe window opened when RStudio is launched

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#7392]
### System details

    RStudio Edition : Desktop
    RStudio Version : v1.4.615
    OS Version      : Windows 10 x64
    R Version       : 4.0.2

### Steps to reproduce the problem

Start RStudio. (In my case, I started RStudio by double-clicking `rstudio.exe` from the view in the Windows File Explorer.)

### Describe the problem in detail

A blank window behind the main window is launched.

### Describe the behavior you expected

No extra window.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### Notes

<img width="1050" alt="Screen Shot 2020-07-20 at 11 33 39 AM" src="https://user-images.githubusercontent.com/1976582/87973098-de25dc00-ca7c-11ea-9198-08120d844f04.png">

**Comment by kevinushey** [comment]
`dumpbin.exe` suggests this is caused by RStudio now being compiled as a Windows CUI application:

```
C:\Program Files\RStudio\bin>dumpbin /headers rstudio.exe
Microsoft (R) COFF/PE Dumper Version 14.16.27034.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file rstudio.exe

PE signature found

File Type: EXECUTABLE IMAGE

FILE HEADER VALUES
            8664 machine (x64)
               6 number of sections
        5F13CF3B time date stamp Sat Jul 18 21:42:35 2020
               0 file pointer to symbol table
               0 number of symbols
              F0 size of optional header
              22 characteristics
                   Executable
                   Application can handle large (>2GB) addresses

OPTIONAL HEADER VALUES
             20B magic # (PE32+)
           14.16 linker version
          40A200 size of code
          255E00 size of initialized data
               0 size of uninitialized data
          385884 entry point (0000000140385884)
            1000 base of code
       140000000 image base (0000000140000000 to 0000000140663FFF)
            1000 section alignment
             200 file alignment
            6.00 operating system version
            0.00 image version
            6.00 subsystem version
               0 Win32 version
          664000 size of image
             400 size of headers
               0 checksum
               3 subsystem (Windows CUI)     <---
< ... >
```

**Comment by kevinushey** [comment]
And this is where we declare the subsystem to be used for Windows Desktop, so it seems like that's somehow being ignored?

https://github.com/rstudio/rstudio/blob/711b96ebaa102b275c4fa966066dc1192acc0f98/src/cpp/desktop/CMakeLists.txt#L318-L320

Alternatively, perhaps we need to be using the "official" CMake API for declaring an executable to require the Windows subsystem. From https://cmake.org/cmake/help/latest/command/add_executable.html:

> If WIN32 is given the property WIN32_EXECUTABLE will be set on the target created. See documentation of that target property for details.

And from https://cmake.org/cmake/help/latest/prop_tgt/WIN32_EXECUTABLE.html#prop_tgt:WIN32_EXECUTABLE:

> When this property is set to true the executable when linked on Windows will be created with a WinMain() entry point instead of just main(). This makes it a GUI executable instead of a console application. See the CMAKE_MFC_FLAG variable documentation to configure use of the Microsoft Foundation Classes (MFC) for WinMain executables. This property is initialized by the value of the CMAKE_WIN32_EXECUTABLE variable if it is set when a target is created.

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.4.747, R Version - 4.0.2, Windows 10 x64 - Desktop 

Unable to reproduce the original issue (although I'm thinking I may have encountered it before it got fixed and not reported it due to thinking it must have been a local issue, need to stop doing that) due to all available [Daily builds](https://dailies.rstudio.com/rstudio/oss/windows/) being post-fix. Tested multiple launches of RStudio on the current daily however (both by launching directly from the .exe file, Project files, and through start menu) and the issue appears fixed.
